src/Sql/Sql.Test/ScenarioTests/ManagedDatabaseCrudScenarioTests.ps1

```diff
@@ -239,6 +239,94 @@ function Test-RestoreManagedDatabase
 	finally
 	{
 		Remove-ResourceGroupForTest $rg
+		Remove-ResourceGroupForTest $rg2
+	}
+}
+
+<#
+	.SYNOPSIS
+	Tests restoring a managed database
+#>
+function Test-RestoreDeletedManagedDatabase
+{
+	# Setup
+	$rg = Create-ResourceGroupForTest
+	$rg2 = Create-ResourceGroupForTest
+	$vnetName = "cl_initial"
+	$subnetName = "Cool"
+
+	# Setup VNET 
+	$virtualNetwork1 = CreateAndGetVirtualNetworkForManagedInstance $vnetName $subnetName $rg.Location
+	$subnetId = $virtualNetwork1.Subnets.where({ $_.Name -eq $subnetName })[0].Id
+
+	$managedInstance = Create-ManagedInstanceForTest $rg $subnetId
+	$managedInstance2 = Create-ManagedInstanceForTest $rg2 $subnetId
+
+	try
+	{
+		# Create with all values
+		$managedDatabaseName = Get-ManagedDatabaseName
+		$collation = "SQL_Latin1_General_CP1_CI_AS"
+		$job1 = New-AzSqlInstanceDatabase -ResourceGroupName $rg.ResourceGroupName -InstanceName $managedInstance.ManagedInstanceName -Name $managedDatabaseName -Collation $collation -AsJob
+		$job1 | Wait-Job
+		$db = $job1.Output
+
+		Assert-AreEqual $db.Name $managedDatabaseName
+
+		$targetManagedDatabaseName1 = Get-ManagedDatabaseName
+		$targetManagedDatabaseName2 = Get-ManagedDatabaseName
+		$targetManagedDatabaseName3 = Get-ManagedDatabaseName
+		$targetManagedDatabaseName4 = Get-ManagedDatabaseName
+		$targetManagedDatabaseName5 = Get-ManagedDatabaseName
+
+		# Wait for 450 seconds for restore to be ready
+		Wait-Seconds 450
```

Comment:
This is an anti-pattern - you should always wait for something instead of waiting for a certain amount time.
In detail, wait a few seconds, go check the status, and wait again if it's not ready, then retry again...
There's common code helping you do this: see [this example](https://github.com/Azure/azure-powershell/blob/master/src/Automation/Automation.Test/ScenarioTests/SourceControlTests.ps1#L117)

Reply:
I agree with you, however backup service is designed in a way where we don't have API to check if backups are taken or not. 

Backup service will work in background and will take log backup every 5 minutes, but customer doesn't have information on those backups, and exactly when they are taken for new database.

Therefore I don't have operation to wait on, the best thing we can do is to rely on our SLA, which is 5 minutes, in which we guarantee backup. This is something we definitively plan on working on in the future and we should revisit these tests once we get APIs for backup service.

Reply:
If so, I suggest add some comments to make it clear

Reply:
Done.